### ISSUE TITLE
Using macros ignores subsequent argument delimiters

### ISSUE THREAD

**Issue by Qix-** [gittup/tup#234]
Consider the following:

```tup
!foo = |> cat -v |>
: foreach * |> !foo "%f" > "%o" |> bin/%B
```

The command that is run for, say, *bar.txt* is `cat -v bar.txt > "%o"` instead of `cat -v bar.txt > "bin/bar"`.

**Comment by VolatileDream** [comment]
Likewise (and most likely related) the following also does not work:
```
!foo = |> echo a >> %o |>
!bar = |> echo b >> %o |>

: |> !foo ; !bar |> output
```

This fails with the output:
```
1) echo a >> output ; !bar
/bin/sh: 1: !bar: not found
 *** tup errors ***
```

The expectation is that both macros are expanded:

`|> echo a >> %o ; echo b >> %o |> output`

**Comment by Qix-** [comment]
Yep. This is causing a lot of issues for us since we're trying to build up a set of rules that handle cross platform tooling...

**Comment by gittup** [comment]
The original issue is fixed by 65fdf0f185c102f17abc4722d714ce4787d59bb7, so %-flags are expanded in the extra command line args.

Is there a use-case where you need to expand multiple !-macros in a single rule? The example given seems like it would be easy to just do:

!foo = |> (echo a; echo b) > %o |>

Not that it wouldn't be nice to have, but unfortunately it's not trivial to implement.
